### PR TITLE
NewsPanel: Remove Use Proxy option and update documentation with recommendations

### PR DESCRIPTION
**Pull request by joshhunt**
**What this PR does / why we need it**:

The `https://cors-anywhere.herokuapp.com` service has been essentially shut down for a while now so the "Use Proxy" option functionally does not work in Grafana, so we're removing it.

If users are trying to show an RSS feed that does not support CORS, they can rehost the feed themselves and/or "manually" prefix the feed URL with a CORS proxy of their liking.

**Which issue(s) this PR fixes**:

Relates to [#37841](https://github.com/grafana/grafana/issues/37841)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -275,6 +275,8 @@ require (
 	github.com/opencontainers/image-spec v1.0.2 // indirect
 	github.com/pierrec/lz4/v4 v4.1.8 // indirect
 	github.com/segmentio/asm v1.1.1 // indirect
+	github.com/tkrajina/go-reflector v0.5.5 // indirect
+	github.com/tkrajina/typescriptify-golang-structs v0.1.7 // indirect
```
Comment (joshhunt) [inline go.mod:279]: ??????? no idea

File: public/app/plugins/panel/news/models.cue [hunk 2]
```diff
@@ -21,7 +21,6 @@ Panel: {
                 PanelOptions: {
                     // empty/missing will default to grafana blog
                     feedUrl?: string
-                    useProxy?: bool
```
Comment (joshhunt) [inline public/app/plugins/panel/news/models.cue:24]: @sdboyer Did I do this right? 👼🏻

**Review by axelavargas — APPROVED** [review]
Code looks good! 🚀

File: docs/sources/visualizations/news-panel.md [hunk 3]
```diff
@@ -10,3 +10,7 @@ weight = 800
 This panel visualization displays an RSS feed. By default, it displays articles from the Grafana Labs blog.
 
 Enter the URL of an RSS in the URL field in the Display section. This panel type does not accept any other queries.
+
+Some RSS feeds may not load if they're not configured to be requested by Grafana's frontend (with the appropriate [CORS headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)). Previous versions of the News panel had a "Use Proxy" option to circumvent this via an external service, which has since been shut down.
```
Comment (achatterjee-grafana) [inline docs/sources/visualizations/news-panel.md:14]:
```suggestion
In version 8.5, we have discontinued the "Use Proxy" option for Grafana news panels. As a result, RSS feeds that are not configured for request by Grafana's frontend (with the appropriate [CORS headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS)) may not load. 
```

File: docs/sources/visualizations/news-panel.md [hunk 4]
```diff
@@ -10,3 +10,7 @@ weight = 800
...
+
+If the RSS feed you're trying to display fails to load, consider rehosting the RSS feed yourself, or prefixing the RSS url with your own "CORS proxy". Alternatively, you can use the community [RSS/Atom data source](https://grafana.com/grafana/plugins/volkovlabs-rss-datasource/) in combination with the [Dynamic text](https://grafana.com/grafana/plugins/marcusolsson-dynamictext-panel/) community panel.
```
Comment (achatterjee-grafana) [inline docs/sources/visualizations/news-panel.md:16]:
```suggestion
If the RSS feed you're trying to display fails to load, consider rehosting the RSS feed or prefixing the RSS URL with your own "CORS proxy". Alternatively, you can use the community [RSS/Atom data source](https://grafana.com/grafana/plugins/volkovlabs-rss-datasource/) in combination with the [Dynamic text](https://grafana.com/grafana/plugins/marcusolsson-dynamictext-panel/) community panel to display the RSS feed.
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added doc review.

**Review by achatterjee-grafana — APPROVED** [review]
LGTM

**Pull request merged**